src/startupscriptgenerator/dotnetcore/scriptgenerator.go

```diff
@@ -102,51 +104,43 @@ func (gen *DotnetCoreStartupScriptGenerator) getStartupCommand() (string, string
 	// as we want to generate a script which does a 'cd' to this directory and run the user startup command
 	publishOutputDir := gen.PublishedOutputPath
 	if publishOutputDir == "" {
-		projDetails := gen.getProjectDetailsAndCache()
-		if projDetails.FullPath == "" {
-			return "", ""
-		}
 
-		publishOutputDir = filepath.Join(projDetails.Directory, OryxPublishOutputDirectory)
+		// Check if current directory is indeed the output directory (Example, like in AppService's case)
+		currentDir := common.GetValidatedFullPath(".")
 
-		logger.LogInformation(
-			"Published output directory not supplied. Checking for default oryx publish output directory at '%s'",
-			publishOutputDir)
-
-		if _, err := os.Stat(publishOutputDir); os.IsNotExist(err) {
-			logger.LogError(
-				"Could not find oryx publish output directory at '%s'. Error: %s",
-				publishOutputDir,
-				err.Error())
-			return "", ""
+		startupFileName := gen.getStartupDllFileName()
+		startupFileFullPath := filepath.Join(currentDir, startupFileName)
+		if _, err := os.Stat(startupFileFullPath); !os.IsNotExist(err) {
```

Comment:
So with this change we check if the current directory has the startup file and if not then fall back to looking for the reserved directory `oryx_publish_output` directory for the startup dll file.